lib/streamlit/config.py

```diff
@@ -128,7 +128,8 @@ def set_user_option(key: str, value: Any) -> None:
         opt = _config_options_template[key]
     except KeyError as ke:
         raise StreamlitAPIException(f"Unrecognized config option: {key}") from ke
-    if opt.scriptable:
+    # Allow e2e tests to set any option
+    if opt.scriptable or get_option("global.e2eTest"):
```

Comment:
@lukasmasuch will be curious on your thoughts here, but I wanted to allow the ability to set options in e2e tests but some are restricted.

Reply:
I think this is fine. There is also an alternative to these variables without the scriptable check:

```python
from streamlit import config

config.set_option(...)
```

Also, I think if a variable is technically able to be set during runtime, we should probably mark it as scriptable. We already have the `hidden` prop to mark internal variables that we don't want to show to users.